ckcp/gitops.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+#quit if exit status of any cmd is a non-zero value
+set -euo pipefail
+# set -x
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" >/dev/null ; pwd)"
+GITOPS_DIR="$(dirname "$SCRIPT_DIR")/gitops"
+KUBECONFIG_CLUSTER="${KUBECONFIG:-$HOME/.kube/config}"
+
+
+
+install_openshift_gitops(){
+  export KUBECONFIG="$KUBECONFIG_CLUSTER"
+
+  local ns="openshift-operators"
+
+  #############################################################################
+  # Install the gitops operator
+  #############################################################################
+  echo -n "  - OpenShift-GitOps: "
+  if ! oc get subscriptions -n "$ns" openshift-gitops-operator >/dev/null 2>&1; then
+    oc apply -f "$GITOPS_DIR/$APP.yaml" --wait >/dev/null
+  fi
+  echo "OK"
+
+  #############################################################################
+  # Wait for the URL to be available
+  #############################################################################
+  echo -n "  - ArgoCD dashboard: "
+  test_cmd="oc get route/openshift-gitops-server --ignore-not-found -n $APP -o jsonpath={.spec.host}"
+  ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  while ! curl --fail --insecure --output /dev/null --silent "https://$ARGOCD_HOSTNAME"; do
+    echo -n "."
+    sleep 2
+    ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  done
+  echo "OK"
+  echo "  - ArgoCD URL: https://$ARGOCD_HOSTNAME"
+
+  #############################################################################
+  # Post install
+  #############################################################################
+
+  # Allow ArgoCD to control resources outside of its namespace
+  oc adm policy add-cluster-role-to-user cluster-admin -z openshift-gitops-argocd-application-controller -n openshift-gitops >/dev/null
+
+  # Log into ArgoCD
+  echo -n "  - ArgoCD Login: "
+  local argocd_password
+  argocd_password="$(oc get secret openshift-gitops-cluster -n openshift-gitops -o jsonpath="{.data.admin\.password}" | base64 --decode --ignore-garbage)"
+  argocd login "$ARGOCD_HOSTNAME" --insecure --username admin --password "$argocd_password" >/dev/null
```

Comment:
add into README.md that argocd is required or create check at the beginning of the script if it's installed.

Reply:
Do we need this command? Suppose that applications will be installed via spec right?

Reply:
> add into README.md that argocd is required or create check at the beginning of the script if it's installed.

Done

Reply:
> Do we need this command? Suppose that applications will be installed via spec right?

The applications will be deployed via a manifest, but we'll be using the argocd cli to get the status of these apps.

Reply:
Possible to do with the oc command -> `oc get apps -n openshift-gitops`

Reply:
We also use the argocd command to add the KCP cluster to ArgoCD. Are you aware of an equivalent `oc` command for [this line](https://github.com/Roming22/pipelines-service/blob/my/ckcp/ckcp/gitops.sh#L175)?

Reply:
ok, it's used yet, so I am doing review only for this MR where the command is not actively used.